core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -419,7 +419,7 @@ private[kafka] class Processor(val id: Int,
     "socket-server",
     metricTags,
     false,
-    ChannelBuilders.serverChannelBuilder(securityProtocol, channelConfigs, credentialProvider.credentialCache))
+    ChannelBuilders.serverChannelBuilder(listenerName, securityProtocol, config, credentialProvider.credentialCache))
```

Comment:
If listenerName is one of the default security protocols, do we need to change it to "" before passing it into serverChannelBuilder() so that we can pick up the right context in JaasConfig?